state/machine.go

```diff
@@ -756,6 +756,15 @@ func (original *Machine) advanceLifecycle(life Life, force bool, maxWait time.Du
 			ContainerIds: containers,
 		}
 	}
+
+	locked, err := original.IsLockedForSeriesUpgrade()
+	if err != nil {
+		return errors.Annotatef(err, "reading machine %s upgrade-series lock", original.Id())
+	}
+	if locked {
+		return errors.Errorf("machine %s is locked for series upgrade", original.Id())
+	}
```

Comment:
If we are doing this check here, we also need to add an assert in the advance lifecycle ops to prevent races

Reply:
Done. TIL about `BeforeHooks`.